extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/QuarkusRecoveryService.java

```diff
@@ -51,4 +51,11 @@ public void create() {
         }
         xaResources.clear();
     }
+
+    @Override
+    public void destroy() {
+        super.destroy();
+        isCreated = false;
+        recoveryManagerService = null;
+    }
```

Comment:
Could you please add a non-regression test for this?

You can find an example of a hot reload test in Hibernate ORM tests:

* Resource: https://github.com/quarkusio/quarkus/blob/ebee0a4b66d1b4dba9c51165e78d22fbf3b90d3e/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/xml/orm/OrmXmlHotReloadTestResource.java#L13
* Test: https://github.com/quarkusio/quarkus/blob/01bee6969efe374d4489d35189c65f34106cea87/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/xml/orm/OrmXmlHotReloadExplicitFileTestCase.java#L13

In your case you'd probably just have a resource that runs an transaction and check that the resource still works after changing it (and automatically triggering a hot reload)

Reply:
I doubt that I will have time today, but do note that the clock is ticking on this fix. But we can merge it now and I can come back to it later though.

Reply:
I agree that the comment about concurrency in particular is not urgent at all.

As to whether we should add non-regression tests when fixing bugs, well, I'll let whoever wants the fix urgently decide.

Reply:
Actually on second thoughts, regression tests ought to be non-negotiable so I will prioritise that ahead of other stuff today, thanks.

Reply:
In the example you gave. what is it that triggers the hot reload?

Reply:
That would be this:

https://github.com/quarkusio/quarkus/blob/01bee6969efe374d4489d35189c65f34106cea87/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/xml/orm/OrmXmlHotReloadExplicitFileTestCase.java#L31-L32

Reply:
In your case you'll probably want to modify some java file instead, maybe the endpoint, to have it return a different value, so that your assertion can check it's been updated?

Reply:
@yrodiere If I added a dummy `application.properties` file in the test/resources directory and changed it via the regression test, would that trigger a reload?

Reply:
You'd need to include it in the resources of your test, like here https://github.com/quarkusio/quarkus/blob/01bee6969efe374d4489d35189c65f34106cea87/extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/xml/orm/OrmXmlHotReloadExplicitFileTestCase.java#L21.

But yes.

I'd be careful that changing the properties changes the behavior of your endpoint though, so that your second assertion checks both that 1. a hot-reload actually occurred and 2. that narayana still works.

Reply:
And do I need to set `quarkus.devservices.enabled` to true in the application.properties that I use with the test?

Reply:
Probably not, since you don't want to test devservices here.

You'll probably want to set the JDBC URL though. But it looks like you use a test profile for that in other tests, not `application.properties`... for some reason (I don't know why): 

https://github.com/quarkusio/quarkus/blob/fd02a43d4b80fe60a3ad5bdaf1e54652c4796bdd/integration-tests/narayana-jta/src/test/java/io/quarkus/narayana/jta/JdbcObjectStoreTestProfile.java 

https://github.com/quarkusio/quarkus/blob/6fc835025144f8d235fb98a94b2e7070bdeba101/integration-tests/narayana-jta/src/test/java/io/quarkus/narayana/jta/TransactionJdbcObjectStoreTest.java#L11

Reply:
I think I've figured it out, I'll let you know if I hit any issues with it ...

Reply:
@mmusgrov The PR build is failing, so I think we'll need to fix that before we merge, at least.
The failure seems unrelated though. I'll try to rebase the PR...

@gsmet I'll let you approve the PR if you're fine with @mmusgrov 's proposal.